File: z/en-de.py
import sys


def main(argv):
    """
    中文Unicode编码，凯撒加密，简单保护私人文件
    """
    if len(argv) > 0:
        file_name, key, is_enc = argv[0], int(argv[1]), True
        if len(argv) == 3 and argv[2][0] == 'd':
            is_enc = False
    else:
        file_name = input("File name: ")
        key = int(input("Secret key number: "))
        enc_or_dec = input("Encode or Decode File(e/d): ")
        is_enc = False
        if enc_or_dec[0] == 'e':
            is_enc = True
    zh_sta, zh_end = 0x4e00, 0x9fa5 + key
    code = ""

    no_code_part = False  # 非加密部分 标志
    no_code_s, no_code_e = 8594, 8592

    with open(file_name, "r", encoding="utf-8") as f:
        for c in f.read():
            if not no_code_part and zh_sta <= ord(c) <= zh_end:
                code += chr(ord(c) + key) if is_enc else chr(ord(c) - key)
            elif ord(c) == no_code_s or ord(c) == no_code_e:
                no_code_part = True if ord(c) == no_code_s else False
                code += c
            else:
                code += c

    pre = "en-" if is_enc else "de-"
    with open(pre + file_name, "w", encoding="utf-8") as f:
        f.write(code)


if __name__ == "__main__":
    main(sys.argv[1:])

File: z/en-de.js
// 访问时恢复加密内容，除了加密文件，好像还可躲审查

window.onload = function(){
    const key = parseInt(prompt("key number:", 0));
    const zh_sta = 0x4e00; 
    const zh_end = 0x9fa5 + key;

    let no_code_part = false  // 非加密部分 标志
    const no_code_s = 8594; 
    const no_code_e = 8592;

    // let root = $("body").firstElementChild;
    let root = document.getElementsByTagName("body")[0].firstElementChild
    function dfs(node){
        if(node.nodeType == 1){
            for(let i=0; i<node.childNodes.length; i++){  // childNodes:包括text
                dfs(node.childNodes[i]) 
            }
        }
        else if(node.nodeType == 3){    // text
            let code = node.textContent,t = "";
            for(let i=0; i<code.length; i++){
                let n = code.charCodeAt(i);
                if(!no_code_part && n >= zh_sta && n <= zh_end){
                    t += String.fromCharCode(n - key);
                }else if(n == no_code_s || n == no_code_e){
                    if(n == no_code_s){
                        no_code_part = true;
                    }else{
                        no_code_part = false;
                    }
                }
                else{
                    t += code[i];
                }
            }
            node.textContent = t;
        }
    }
    dfs(root);
}